### PR TITLE
Drop support for Python 2 and allow Pandas 1.x

### PR DESCRIPTION
**Pull request by mikesaelim**
Main points:
* Drop support for Python 2
* Upgrade civis-jupyter-extensions to `~=1.0`, which allows Pandas 1.x
* Require civis `~1.9` to remove deprecation warnings
* Prepare for civis-jupyter-notebook v2.0.0 release

### CODE REVIEW

File: CHANGELOG.md
Comment (mikesaelim) [inline CHANGELOG.md:32]: This item had been released in v1.0.1 but wasn't included in the changelog for that version.

File: README.rst
Comment (mikesaelim) [inline README.rst:14]: In this and the example Dockerfile, we should have been setting this environment variable earlier to avoid an error when running `civis-jupyter-notebooks-install`.

File: civis_jupyter_notebooks/assets/civis_client_config.py
Comment (mikesaelim) [inline civis_jupyter_notebooks/assets/civis_client_config.py:7]: I also removed the deprecated `resources='all'` argument from our instantiations of the API client.  It was deprecated in civisanalytics/civis-python#243.

File: .travis.yml
Comment (mikesaelim) [reply .travis.yml:21]: Yeah, it's saying only to run this test for the Python 3.7 part of the build matrix.  The test itself doesn't depend on the Travis Python version - it builds a Docker image with this code in it - so I think the conditional is trying to have the test run only once.

File: README.rst
Comment (mikesaelim) [reply README.rst:14]: Just that `DEFAULT_KERNEL` wasn't defined.

File: civis_jupyter_notebooks/assets/civis_client_config.py
Comment (mikesaelim) [reply civis_jupyter_notebooks/assets/civis_client_config.py:7]: Ah yeah that's a good point.  I'll revert this change, because there's no real reason to break compatibility with `civis` versions before 1.9.

Comment (leanne73) [reply civis_jupyter_notebooks/assets/civis_client_config.py:7]: Hmm, [1.9](https://github.com/civisanalytics/civis-python/releases/tag/v1.9.0) itself is already over 2 years old. Maybe a question for python client maintainers, but do we really want to keep supporting versions that old? I imagine there are plans to fully remove the `resources` parameter, given the deprecation warnings, so we will probably need to make this change eventually

File: README.rst
Comment (leanne73) [reply README.rst:14]: ah, that makes sense

File: .travis.yml
Comment (leanne73) [reply .travis.yml:21]: Oh, I see. That's a little odd, but makes sense.

File: civis_jupyter_notebooks/assets/civis_client_config.py
Comment (mikesaelim) [reply civis_jupyter_notebooks/assets/civis_client_config.py:7]: Yeah, I agree, we should ask the maintainers.  I think it's out of scope for this PR though.

Comment (mikesaelim) [reply civis_jupyter_notebooks/assets/civis_client_config.py:7]: Discussed this offline; we'll bump the required version to 1.9.

**Review by mheilman — COMMENTED** [review]
No suggestions from me. I'll let Leanne approve.

**Review by leanne73 — DISMISSED** [review]
LGTM

**Review by leanne73 — APPROVED** [review]
LGTM

**Pull request merged**
